virttest/qemu_storage.py

```diff
@@ -561,7 +561,10 @@ def _backing_access_secrets(self):
     def _secret_objects(self):
         """All secret objects str needed for command line."""
         secret_objects = self.encryption_config.image_key_secrets
-        secret_obj_str = "--object secret,id={s.aid},data={s.data}"
+        if self.params.get("secret_obj_file") == "yes":
```

Comment:
if we introduce a new param, just like Xu Han said, we'd better add the param's namespace, here secret_obj_file is used for image, so image_secret_obj_file might be better. Also, it's better to add it and comment it in shared/cfg/guest-hw.cfg, under image_secret.

Here is another concern from curl backend, just FYI.
You know, for http/https, we can pass cookie with a secret object, it also uses the format of '--object secret, data=xxxx' to define the secret object, but when we associate the secret object with an image in command line, we have to use 'cookie-secret=sec_id' instead of 'key-secret=id' or 'password-secret=id', what I mean is, although when defining a luks format, we now only use either 'key-secret' or 'password-secret', but we might introduce a new 'xxx-secret' in future, just like 'cookie-secret' for http, at that time, a flag param 'secret_obj_file' cannot cover all scenarios(it only covers two: yes--'password-secret', no----'key-secret'), so what about introducing a param like 'image_secret_type', now it has two values: 'key' and 'password', and it defaults to 'key' if it's not defined. we can easily enhance it a new value 'xxxx', used for 'xxxx-secret' in future.

Reply:
updated, many thanks.